autosubmit/autosubmit.py

```diff
@@ -2875,48 +2875,52 @@ def statistics(expid, filter_type, filter_period, file_format, section_summary,
         return True
 
     @staticmethod
-    def clean(expid, project, plot, stats):
-        """
-        Clean experiment's directory to save storage space.
+    def clean(expid: str, project: bool, plot: bool, stats: bool) -> bool:
+        """Clean the experiment directory to save storage space.
+
         It removes project directory and outdated plots or stats.
 
-        :type plot: bool
+        When it deletes plots or stats, it keeps the two newest files
+        (i.e. two newest statistics files, and two newest plot files).
+
+        :param expid: The experiment ID.
+        :type: str
+        :param project: Whether to clean the experiment project folder or not.
         :type project: bool
-        :type expid: str
+        :param plot: Whether to delete plot files (keeping two newest) or not.
+        :type plot: bool
+        :param stats: Whether to delete statistics files (keeping two newest) or not.
         :type stats: bool
-        :param expid: identifier of experiment to clean
-        :param project: set True to delete project directory
-        :param plot: set True to delete outdated plots
-        :param stats: set True to delete outdated stats
+        :return: ``True`` is the command ran successfully and ``False`` otherwise.
+        :rtype: bool
+        :raises: AutosubmitCritical if anything goes wrong cleaning the experiment folders.
         """
-        from .monitor.monitor import Monitor
+        from .monitor.monitor import clean_plot, clean_stats
 
         try:
-            exp_path = os.path.join(BasicConfig.LOCAL_ROOT_DIR, expid)
-
             if project:
-                autosubmit_config = AutosubmitConfig(
-                    expid, BasicConfig, YAMLParserFactory())
+                autosubmit_config = AutosubmitConfig(expid, BasicConfig, YAMLParserFactory())
                 autosubmit_config.check_conf_files(False)
 
                 project_type = autosubmit_config.get_project_type()
                 if project_type == "git":
                     Log.info("Registering commit SHA...")
                     autosubmit_config.set_git_project_commit(autosubmit_config)
-                    autosubmit_git = AutosubmitGit(expid[0])
+                    autosubmit_git = AutosubmitGit(expid)
```

Comment:
Bug :bug: , as it was using the first character of the `expid` (other functions here were using `expid` as a `str`. Maybe this was copied from some other command that accepts multiple expids?)